pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -557,14 +557,27 @@ func (p *testPlanner) systemSetupSteps() []testStep {
 	}
 
 	setupContext := p.nonUpgradeContext(initialVersion, SystemSetupStage)
+	var clusterStartSteps []testStep
+
+	clusterInitHooks := p.hooks.ClusterInitSteps(setupContext, p.prng)
+	if len(clusterInitHooks) > 0 {
+		clusterStartSteps = append(clusterStartSteps, clusterInitHooks...)
+	}
+	// N.B. Add the cluster start step after user hooks as the framework will run liveness
+	// checks once the start step is run, which is incompatible with cluster init hooks.
+	clusterStartSteps = append(clusterStartSteps,
+		p.newSingleStepWithContext(setupContext,
+			startStep{
+				version:            initialVersion,
+				rt:                 p.rt,
+				initTarget:         p.currentContext.System.Descriptor.Nodes[0],
+				waitForReplication: p.shouldWaitForReplication(),
+				settings:           p.clusterSettingsForSystem(initialVersion),
+			}),
+	)
+
+	steps = append(steps, p.concurrently(clusterInitLabel, clusterStartSteps)...)
```

Comment:
`p.concurrently` will break the "happens before" ordering wrt clusterInitHooks.

Reply:
Hmm, yeah especially with delayed steps that would break the ordering even more. Okay, I've changed `clusterInitHook` to be `beforeClusterStartHook` instead. It works better for this use case to guarantee  it's spun up before cluster start and is imo more generalized for other potential use cases anyway.

Or do you mean that the clusterInitHooks should be run sequentially? I don't have an issue with that, I was just copying what `AfterTestHooks` and `AfterFinalizationHooks` does.